operators/dream_texture.py

```diff
@@ -96,8 +96,14 @@ def image_writer(shared_memory_name, seed, width, height, upscaled=False):
                     if area.type == 'IMAGE_EDITOR':
                         area.spaces.active.image = image
                 scene.dream_textures_prompt.seed = str(seed) # update property in case seed was sourced randomly or from hash
+                # create a hash from the Blender image datablock to use as unique ID of said image and store it in the prompt history
+                # and as custom property of the image. Needs to be a string because the int from the hash function is too large
+                image_hash = str(hash(image))
```

Comment:
If you use hashlib to create a sha256 hash you can find the image even if it doesn't have the `dream_textures_hash` property, because it's consistent across runs.

```suggestion
                image_hash = hashlib.sha256((np.array(image.pixels) * 255).tobytes()).hexdigest()
```
Then compare against the hash again when reloading.

That may be too slow though, if so I'm ok merging as is.

Reply:
If I understand correctly, the hash would need to be re-computed for on average half of the existing image datablocks when recalling the prompt. Doesn't that require lots of memory access?

Reply:
I’m ok requiring the custom property on the image datablock. I’d still prefer using a sha256 hash of the image pixels because it will provide a consistent value in case we want to use it for other purposes, e.g. integration tests.

Reply:
I'm agnostic to the hash method 🙂

Reply:
My only issue with `hash(...)` is that it is not consistent across runs of Python afaik.